### ISSUE TITLE
Please remove legacy VoIP feature

### ISSUE THREAD

**Issue by k-gruenberg** [dinhvh/libetpan#444]
[In these two lines](https://github.com/dinhvh/libetpan/blob/5c9eb6b6ba64c4eb927d7a902317410181aacbba/src/data-types/mailstream_cfstream.c#L495) you are referencing `kCFStreamNetworkServiceTypeVoIP`. According to [this thread](https://forums.developer.apple.com/forums/thread/738799) however, it is referring to a legacy VoIP feature that has finally been removed in iOS 17. The result is that no matter whether I use the [MailCore](https://github.com/MailCore/mailcore2) or the [Postal](https://github.com/snipsco/Postal) library, I get the exact same crash when trying to connect to an IMAP server. The fact that you surrounded the whole thing with `#if TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR` made this especially hard to debug as the error does not occur when running my app in the simulator. Please remove the references to this feature as it was non-functional for a long time now anyways. Thanks!

**Comment by barracuda156** [comment]
@k-gruenberg It is _only_ used for iOS, not macOS, is it?

**Comment by k-gruenberg** [comment]
@barracuda156 I think that's pretty much what `#if TARGET_OS_IPHONE` says, yeah. Personally, I'm developing an iOS-only app (for now) and that's where I wanted to use it. If you're interested in my current solution to the problem: use a working Rust library instead and use FFI to call the Rust code from Swift. Pretty pathetic for a language like Swift when there aren't really any working email libraries...

**Comment by barracuda156** [comment]
> that's pretty much what `#if TARGET_OS_IPHONE` says

@k-gruenberg Yeah, that’s what I also thought, just wanted to make sure. My concern is to avoid breaking the build for older macOS versions, but anything iOS-only is inconsequential for that.

**Comment by dinhvh** [comment]
Addressed via https://github.com/dinhvh/libetpan/commit/22c96e6ab708332872a3722e7d718d88c35d0357
